### DISCUSSION TITLE
Add Rope gun

### DISCUSSION THREAD

**Discussion by metalgearsloth** [space-wizards/space-station-14#4047] [Feature Request]
Probably best to wait until named joints are in to reduce their overall JANK but we talked about this on discord a little bit. Best to get it working first then it can be tuned later.

Essentially, when fired:
* Raycast out like hitscan
* (if valid) form DistanceJoint between yourself and the target with a low length so you get pulled in(?)
* Firing again detaches (or dropping it, or if LoS breaks with the occasional raycast, etc.)

Max distance probably needs to be infinite

**Comment by RemieRichards** [comment]
That's a grappling hook, not a rope gun.

**Comment by metalgearsloth** [comment]
Okay ms. smartypants with ur big words

**Comment by RemieRichards** [comment]
I just think like, actual ropes, for traversing the outside of the station safely should definitely be a thing, and *that'd* be more akin to a rope gun than this :P
